Code/Tools/ProjectManager/Source/ProjectExportWorker.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ProjectExportWorker.h>
+#include <ProjectManagerDefs.h>
+#include <PythonBindingsInterface.h>
+#include <ProjectUtils.h>
+
+#include <QDir>
+#include <QFile>
+#include <QProcess>
+#include <QProcessEnvironment>
+#include <QTextStream>
+#include <QThread>
+
+namespace O3DE::ProjectManager
+{
+    ProjectExportWorker::ProjectExportWorker(const ProjectInfo& projectInfo)
+        : QObject()
+        , m_projectInfo(projectInfo)
+    {
+    }
+
+    void ProjectExportWorker::ExportProject()
+    {
+        auto result = ExportProjectForPlatform();
+
+        if (result.IsSuccess())
+        {
+            emit Done();
+        }
+        else
+        {
+            emit Done(result.GetError());
+        }
+    }
+
+    AZ::Outcome<QString, QString> ProjectExportWorker::GetLogFilePath() const
+    {
+        QDir logFilePath(m_projectInfo.m_path);
+        // Make directories if they aren't on disk
+        if (!logFilePath.cd(ProjectBuildPathPostfix))
+        {
+            bool madeDirectory = logFilePath.mkpath(ProjectBuildPathPostfix);
+            if (!madeDirectory)
+            {
+                return AZ::Failure("Unable to make log directory for project build path");
+            }
```

Comment:
You don't really need a variable, and you should add the path to the error message.

```suggestion
            if (!logFilePath.mkpath(ProjectBuildPathPostfix))
            {
                return AZ::Failure(AZStd::string::format("Unable to create log directory '%s' for project build path", logFilePath.absoluteFilePath(ProjectBuildPathPostfix).toUtf8().constData()));
            }
```

Reply:
The proposed change causes function signature mismatch between `AZ::Outcome<QString, QString>` and `AZStd::unexpected<E>`. 

The next best thing I did was build the error message as a QString and pass that to AZ::Failure. That passed the function signature check.